block/block.go

```diff
@@ -148,3 +149,30 @@ func (blockchain *Blockchain) Extend(commitToNextBlock Commit) {
 	blockchain.blocks[commitToNextBlock.Polka.Block.Header] = commitToNextBlock
 	blockchain.head = commitToNextBlock
 }
+
+// calculateHeader will return the SHA3 hash of the parentHeader, timestamp, round,
+// height, and transactions.
+// TODO: (Review) Should block header include timestamp as well, given that `Equal`
+// ignores Time when checking for equality between 2 blocks. (See comment in `Equal`)
+func calculateHeader(block Block) [32]byte {
+	headerString := fmt.Sprintf("Block(ParentHeader=%s,Timestamp=%s,Round=%d,Height=%d,Transactions=[", base64.StdEncoding.EncodeToString(block.ParentHeader[:]), block.Time.String(), block.Round, block.Height)
+
+	isFirstTx := true
+	for _, tx := range block.Txs {
+		data, err := tx.Marshal()
+		if err != nil {
+			// FIXME: handle this error
+			fmt.Printf("[calculateHeader] error marshalling transaction: %v\n", err)
+			continue
+		}
+		if isFirstTx {
+			headerString = fmt.Sprintf("%s%s", headerString, base64.StdEncoding.EncodeToString(data))
+			isFirstTx = false
+			continue
+		}
+		headerString = fmt.Sprintf("%s,%s", headerString, base64.StdEncoding.EncodeToString(data))
```

Comment:
It is important to consider the scaling of what you have written here. As we allow hundreds of transactions in a block, this header string becomes unreadable and therefore fairly useless. It is better to have a transaction header that is the SHA3 hash of all the transactions:

```go
txHeader := make([]byte, 32 * len(block.Txs))
for i, tx := range block.Txs {
    txHeader := tx.Header() // Method on the Transaction interface that returns sig.Hash
    copy(txHeaders[32 * i:], txHeader[:])
}
txHeaderSHA3 := sha3.Sum256(txHeader)
txHeaderB64 := base64.StdEncoding.EncodeToString(txHeaderSHA3)
return headerString := fmt.Sprintf("Block(ParentHeader=%s,Timestamp=%s,Round=%d,Height=%d,TxHeader=%s)", base64.StdEncoding.EncodeToString(block.ParentHeader[:]), block.Time.String(), block.Round, block.Height, txHeaderB64)
```

Reply:
Also, also, the timestamp should be seconds since UNIX epoch. Not formatted as a string.